dialoptions.go

```diff
@@ -310,6 +312,8 @@ func WithBlock() DialOption {
 //
 // Notice: This API is EXPERIMENTAL and may be changed or removed in a
 // later release.
+// For more information about this anti-pattern
+// see https://github.com/grpc/grpc-go/blob/master/Documentation/anti-patterns.md.
```

Comment:
Please move this above the `// # Experimental` line and use the same text as above.  (and similar request below)